lib/Dialect/ESI/ESIPasses.cpp

```diff
@@ -440,31 +453,32 @@ bool ESIPortsPass::updateFunc(HWModuleOp mod) {
       newArgNames.push_back(argNameAttr);
       continue;
     }
-
     // When we find one, add a data and valid signal to the new args.
-    newArgTypes.push_back(chanTy.getInner());
+    Value data;
+    if (chanTy.hasData()) {
+      newArgTypes.push_back(chanTy.getInner());
+      newArgNames.push_back(argNameAttr);
+      data = mod.front().insertArgument(blockArgNum++, chanTy.getInner(),
+                                        mod.getArgument(argNum).getLoc());
+    } else {
+      // This is a data-less channel - need to create a none-typed SSA value
+      // to feed into wrapvr.
+      data = modBuilder.create<esi::NoneSourceOp>();
+    }
+
     newArgTypes.push_back(i1);
-    newArgNames.push_back(argNameAttr);
     newArgNames.push_back(appendToRtlName(argNameAttr, "_valid"));
-    // Add the BlockArguments.
-    Value data = mod.front().insertArgument(blockArgNum, chanTy.getInner(),
-                                            mod.getArgument(argNum).getLoc());
-    Value valid = mod.front().insertArgument(blockArgNum + 1, i1,
+    Value valid = mod.front().insertArgument(blockArgNum++, i1,
```

Comment:
nit: I don't like using `++` or `--` in anything but *very* simple expressions. The inline side-effect it has makes skimming code more difficult when debugging. "Where the fuck did this variable get incremented?" has bit me a number of times in the past.

Making the increment explicit (like was there) and only at one location is (IME) more debug-able.

Reply:
To me, inline `++`, `--` do have their place in code like this - control-dependent modifications of a single variable. But i can see that this may be too far into personal-preferred-style territory and not suitable for OSS like this.